### PR TITLE
Vector tweak changes

### PR DESCRIPTION
**Pull request by vshekar**
Set of changes that updates a vectors start and end points correctly when they are moved in a 2D plane

### CODE REVIEW

File: daq_utils.py [hunk 1]
```diff
@@ -124,6 +124,28 @@ def wave2energy(w, digits=2):
   else:
     return float(f"%.{digits}f" % (EV_ANGSTROM_CONSTANT/w))
 
+def gonio2lab(x_gonio, y_gonio, z_gonio, omega_deg):
```
Comment (JunAishima) [inline daq_utils.py:127]: does this follow one of NSLS-II or beamline conventions for x, y, z?

File: lsdcGui.py [hunk 2]
```diff
@@ -4714,30 +4714,90 @@ def removePuckCB(self):
       dewarPos, ok = DewarDialog.getDewarPos(parent=self,action="remove")
       
 
-    def getVectorObject(self):
+    def transform_vector_coords(self, prev_coords, current_raw_coords):
```
Comment (JunAishima) [inline lsdcGui.py:4705]: good documentation of complex functions!

File: lsdcGui.py [hunk 3]
```diff
@@ -4749,34 +4809,19 @@ def drawVector(self):
...
-      try:
-        if self.vecLine != None:
-          self.scene.removeItem(self.vecLine)
-      except AttributeError: # likely due to self.vecLine not defined yet
-        pass
-
+      if self.vecLine:
+        self.scene.removeItem(self.vecLine)
+        self.vecLine = None
+
```
Comment (JunAishima) [inline lsdcGui.py:4820]: do we know whether we are still getting these `AttributeError`s or not? (several areas of this commit) we may get more crashes of the GUI

**Review by JunAishima — DISMISSED** [review]
Generally, it is good to see some refactoring to make this code easier to read while adding the new functionality.

File: lsdcGui.py (hunk 3 above)
Comment (vshekar) [reply lsdcGui.py:4820]: We shouldn't get attribute error for this specific variable as it is initialized to None: [here](https://github.com/NSLS-II/lsdc/blob/6dce1d9c5927b1b41394823e48526887458b8bd1/lsdcGui.py#L2203). So it should be safe to check its value anywhere in the app

File: daq_utils.py (hunk 1 above)
Comment (vshekar) [reply daq_utils.py:127]: Unsure what the NSLS-II conventions are, I think it uses beamline conventions. X is the direction of the sample holder, Y is up and down and Z should be in the direction of the beam (sorry for the poor description)

File: lsdcGui.py (hunk 3 above)
Comment (JunAishima) [reply lsdcGui.py:4820]: you're right, as long as it is defined, we shouldn't need that exception. resolving this.

File: daq_utils.py (hunk 1 above)
Comment (JunAishima) [reply daq_utils.py:127]: now that I think about it, I think this should be added to the general documentation, not what you are working on.

resolving

**Pull request merged**